modules/vebal/voting-list.service.ts

```diff
@@ -0,0 +1,188 @@
+    return fetchL2GaugesInfo(poolIds, network);
+}
+
+async function fetchMainnetGaugesInfo(poolIds: string[]): Promise<GaugesInfo> {
+    const gaugesInfo = await gaugeSubgraphService.getGaugesInfo(poolIds);
+    console.log(`⬇️  Fetched ${gaugesInfo.length} gauge info records for Mainnet`);
+    return gaugesInfo.map((rootGauge) => {
+        return { ...rootGauge, chain: 'MAINNET' };
+    });
+}
+
+async function fetchL2GaugesInfo(poolIds: string[], network: NetworkConfig): Promise<GaugesInfo> {
+    const streamers = (await gaugeSubgraphService.getGaugesStreamers(poolIds)).map((gauge) => {
+        return {
+            address: gauge.id,
+            streamer: gauge.streamer,
+            poolId: gauge.poolId as string,
+        };
+    });
+
+    const poolIdsByRecipient = streamers.reduce((acc: Record<string, string>, streamer) => {
+        const recipient = streamer.streamer ? streamer.streamer : streamer.address;
+        acc[recipient] = streamer.poolId;
+        return acc;
+    }, {});
+
+    // TODO: improve TS check
+    const chainName: SubgraphChain = capitalize(network.data.chain.slug) as SubgraphChain;
+
+    const recipients = Object.keys(poolIdsByRecipient);
+    let rootGauges = await gaugeSubgraphService.getRootGaugesInfo(recipients, chainName);
+    console.log(`⬇️  Fetched ${rootGauges.length} gauge info records for ${chainName}`);
+
+    return rootGauges.map((rootGauge) => {
+        // Manually add poolId as we cannot get it in the RootGauges query above
+        return { ...rootGauge, poolId: poolIdsByRecipient[rootGauge.recipient], chain: network.data.chain.prismaId };
+    });
+}
+
+/**
+ * Excludes killed gauges with zero weight
+ */
+async function filterValidGauges(gauges: GaugesInfo): Promise<GaugesInfo> {
+    const killedGaugesList = gauges.filter(({ isKilled }) => isKilled).map(({ id }) => getAddress(id));
+
+    const killedGaugesWeight = await getGaugeRelativeWeight(killedGaugesList);
+
+    const validGauges = gauges.filter(({ id, isKilled }) => !isKilled || killedGaugesWeight[getAddress(id)] !== '0.0');
+    return validGauges;
+}
+
+export async function getGaugeRelativeWeight(gaugeAddresses: string[]): Promise<Record<string, string>> {
+    //TODO: Include this address in network config
+    const veBALHelperAddress = '0x8e5698dc4897dc12243c8642e77b4f21349db97c';
+    const multicall = getContractAt(networkContext.data.multicall3, multicall3Abi);
+
+    const veBalHelpers = new Interface(veBalHelpersAbi);
+
+    const calls = gaugeAddresses.map((address) => [
+        veBALHelperAddress,
+        false, // do not allow failures
+        veBalHelpers.encodeFunctionData('gauge_relative_weight', [address]),
+    ]);
+    // const result = await multicall.callStatic.aggregate3(calls);
+    // const weights = mapValues(result, weight => formatUnits(weight, 18));
+
+    //DEBUG: Return fake results until multicall is tested
+    return gaugeAddresses.reduce((acc: Record<string, string>, address: string) => {
```

Comment:
For now, I use fake data to avoid using the real multicall. I will do that once we decide the overall design.